runtime/src/accounts_db.rs

```diff
@@ -1958,51 +1963,56 @@ impl AccountsDb {
-            .iter()
-            .filter(|(pubkey, stored_account)| {
-                let lookup = if is_startup {
-                    self.accounts_index.get_account_read_entry_with_lock(
-                        pubkey,
-                        accounts_index_map_lock_ref.unwrap(),
-                    )
-                } else {
-                    self.accounts_index.get_account_read_entry(pubkey)
-                };
+        for (pubkey, stored_account) in &stored_accounts {
+            let lookup = if is_startup {
+                self.accounts_index
+                    .get_account_read_entry_with_lock(pubkey, accounts_index_map_lock_ref.unwrap())
+            } else {
+                self.accounts_index.get_account_read_entry(pubkey)
+            };
 
-                if let Some(locked_entry) = lookup {
-                    let is_alive = locked_entry.slot_list().iter().any(|(_slot, i)| {
-                        i.store_id == stored_account.store_id
-                            && i.offset == stored_account.account.offset
-                    });
-                    if !is_alive {
-                        // This pubkey was found in the storage, but no longer exists in the index.
-                        // It would have had a ref to the storage from the initial store, but it will
-                        // not exist in the re-written slot. Unref it to keep the index consistent with
-                        // rewriting the storage entries.
-                        locked_entry.unref()
-                    } else {
-                        alive_total += stored_account.account_size as u64;
-                    }
-                    is_alive
+            if let Some(locked_entry) = lookup {
+                let is_alive = locked_entry.slot_list().iter().any(|(_slot, i)| {
+                    i.store_id == stored_account.store_id
+                        && i.offset == stored_account.account.offset
+                });
+                if !is_alive {
+                    // This pubkey was found in the storage, but no longer exists in the index.
+                    // It would have had a ref to the storage from the initial store, but it will
+                    // not exist in the re-written slot. Unref it to keep the index consistent with
+                    // rewriting the storage entries.
+                    unrefed_pubkeys.push(pubkey);
+                    locked_entry.unref();
                 } else {
-                    false
+                    alive_total += stored_account.account_size as u64;
+                    alive_accounts.push((pubkey, stored_account));
                 }
-            })
-            .collect();
+            }
+        }
         drop(accounts_index_map_lock);
         index_read_elapsed.stop();
         let aligned_total: u64 = self.page_align(alive_total);
 
+        // If not saving maps or more than 1 page of data, then skip the shrink
```

Comment:
What does 'maps' mean here? And just clarifying this should be 'maps or' vs 'maps of'?

Reply:
memory maps